lesson4/index.md

```diff
@@ -0,0 +1,247 @@
+---
+title: Recap loops
+nav_order: 4
+has_children: true
+---
+
+# Lesson 4: Recap loops
+
+## Goals
+* Get to know Sunday and Flo
+* Recap `for`
+* Recap `while`
+* Recap `do` ... `while`
+* Recap `for` each
+* Code, code, code 🤩
+
+## Recap & Homework check
+Let's look into the homework from lesson 3
+
+## for loops
+For loops are the most commonly used loops in Java.
+
+Their structure is as follows:
+```java
+for (<INITIALIZATION>; <CONDITION>; <INCREMENT>) {
+    // <CODE TO EXECUTE>
+}
+```
+
+For loops are usually used to run `<CODE TO EXECUTE>` a given amount of times.
+They are using an `INITIALIZATION` statement which is run exactly once at the beginning, 
+a `CONDITION` statement which is run after each iteration and 
```

Comment:
condition is run before the iteration

Reply:
Correct. Fixing

Reply:
Would be good to point out that the for loop stops once the condition evaluates to false.